ui/pages/confirmations/hooks/useSimulationMetricsNoShow.ts

```diff
@@ -0,0 +1,35 @@
+import { useSelector } from 'react-redux';
+import {
+  UseSimulationMetricsProps,
+  useSimulationMetrics,
+} from '../components/simulation-details/useSimulationMetrics';
+import { selectUseTransactionSimulations } from '../selectors/preferences';
+
+/**
+ * This hook is to capture simulation metrics in the case where the simulation UI is not visible
+ * e.g. when a send transaction is initiated through the wallet itself.
+ * (only exception is if the user had disabled it themselves).
+ *
+ * @param options0
+ * @param options0.enableMetrics
+ * @param options0.transactionId
+ */
+export function useSimulationMetricsNoShow({
+  enableMetrics,
+  transactionId,
+}: {
+  enableMetrics: boolean;
+  transactionId: string;
+}) {
+  const useTransactionSimulations = useSelector(
+    selectUseTransactionSimulations,
+  );
+
+  useSimulationMetrics({
+    enableMetrics: useTransactionSimulations && enableMetrics,
+    balanceChanges: [],
```

Comment:
I'm not sure I understand what this is achieving.

If we're not providing actual balance changes and simulation data, then how can we generate the correct metrics for the transaction?

I understood the scope of this fix was to still perform simulation as before, and gather the real metrics using the real balance changes, but simply not render the component for the user?

Reply:
I agree to field to distinguish `no show` will be useful.